Yosemite/Yosemite/Tools/Cache/CodablePersistentCache.swift

```diff
@@ -0,0 +1,80 @@
+import Foundation
+
+struct CodableCacheEntry<T: Codable>: Codable {
+    let timestamp: Date
+    let value: T
+    let timeToLive: TimeInterval
+
+    init(value: T, timeToLive: TimeInterval) {
+        self.timestamp = .now
+        self.value = value
+        self.timeToLive = timeToLive
+    }
+
+    var isValid: Bool {
+        return Date().timeIntervalSince(timestamp) < timeToLive
+    }
+}
+
+// A class to handle persistent caching of Codable objects tied to a Site
+class CodablePersistentCache<T: Codable> {
+    private let cacheDirectory: URL
+    private let fileManager: FileManager
+
+    init() {
+        self.fileManager = FileManager.default
```

Comment:
If you inject `FileManager` into this method, you can mock it in tests ( ͡° ͜ʖ ͡°)

Reply:
Thanks for the tip Will!